### ISSUE TITLE
Client metrics reporting configurations for 5GMS lack unique identifier

### ISSUE THREAD

**Issue by rjb1000** [5G-MAG/Standards#46]
# Context
The M1 Metrics Reporting Provisioning API specified in clause 7.8 of TS 26.512 was expanded to allow multiple different QoE metrics reporting configurations to be provisioned, potentially reporting different metrics or using different metrics reporting schemes. Each `MetricsReportingConfiguration` resource provisioned at reference point M1 is assigned a resource identifier – the Metrics Reporting Configuration Identifier – that is unique within the scope of its parent Provisioning Session Identifier.

All metrics reporting configurations for a given Provisioning Session are included in the Service Access Information for that Provisioning Session that is supplied by the 5GMS AF to the Media Session Handler at reference point M5 (or else passed privately to the 5GMS-Aware Application via M8), as specified in clause 11.2.3.1 of TS 26.512.

When the Media Session Handler wants to submit a metrics report to the 5GMS AF, it uses the Metrics Reporting API specified in clause 11.4 of TS 26.512. The HTTP request specified in clause 11.4.2 requires both the Provisioning Session Identifier and the Metrics Reporting Configuration Identifier to be cited as path elements in the request URL.

# Problem description
The `ServiceAccessInformation.clientMetricsReportingConfiguration` resource does not include the `metricsReportingConfigurationId` needed by the Media Session Handler to invoke the Metrics Reporting API, as required by clause 11.4.2 of TS 2.512.

# Proposed remedy
Add the missing parameter as `ServiceAccessInformation.clientMetricsReportingConfiguration.metricsReportingConfigurationId` in table 11.2.3.1-1 of TS 26.512.

**Comment by gheikkila** [comment]
Some thoughts now when I started to write a CR. The SAI doesn't contain the metricsReportingConfigurationId, but it does contain the serverAddresses array. You **could** interpret this list of URL:as as being "complete", i.e. already formatted according to 11.4.2, which then would enable reporting without explicitly knowing the metricsReportingConfigurationId.

On the other hand, the text in 11.4.2 specifically states that the Client shall "substitute" the template string with the actual  metricsReportingConfigurationId, which to me indicates that the serverAddresses would only specify the apiRoot part?

![image](https://user-images.githubusercontent.com/7382690/211763193-52993734-10b4-419f-84cf-46356bef3023.png)

**Comment by gheikkila** [comment]
Contribution to 3GPP sent in [https://www.3gpp.org/ftp/TSG_SA/WG4_CODEC/3GPP_SA4_AHOC_MTGs/SA4_MBS/Docs/S4aI230026.zip](url)

**Comment by rjb1000** [comment]
> The SAI doesn't contain the metricsReportingConfigurationId, but it does contain the serverAddresses array. You could interpret this list of URL:as as being "complete", i.e. already formatted according to 11.4.2, which then would enable reporting without explicitly knowing the `metricsReportingConfigurationId`.
>
> On the other hand, the text in 11.4.2 specifically states that the Client shall "substitute" the template string with the actual `metricsReportingConfigurationId`, which to me indicates that the `serverAddresses` would only specify the `apiRoot` part?

I think you have correctly interpreted the current specification, @gheikkila. But it's an interesting thought that we could provide a simpler remedy to this problem by redefining the `serverAddresses` array to contain complete URLs, including the `metricsReportingConfigurationId` as well as the `provisioningSessionId`. This would engender a greater degree of opacity at reference point M5: the Media Session Handler would no longer need to understand the structure of the URL. (This might be considered a desirable or undesirable outcome and I'm interested to get a view from @ibouazizi on that.)

In the interests of API consistency, we would probably want to make an equivalent change in the Service Access Information resource type for the *consumption reporting endpoint addresses* and the *network assistance endpoint address* (which should probably be changed to an array for consistency). For the *dynamic policy invocation endpoint addresses*, the same change could be made too, but note that the Media Session Handler would still need to populate the `{dynamicPolicyId}` path element itself.

Let's discuss on tomorrow's joint 5G-MAG/SA4 call.

**Comment by rjb1000** [comment]
> Contribution to 3GPP sent in [S4aI230026](https://www.3gpp.org/ftp/TSG_SA/WG4_CODEC/3GPP_SA4_AHOC_MTGs/SA4_MBS/Docs/S4aI230026.zip)

Thanks, @gheikkila. I note that this covers Rel-16 only, and we will eventually need a Rel-17 mirror of this change too.

Also, we will need to add OpenAPI changes in annex C once a way forward is agreed.

My plan is to aggregate all TS 26.512 changes involving the M1 and M5 APIs into a pair of jumbo CRs (once each for Rel-16 and Rel-17) for agreement at SA4#122.

**Comment by gheikkila** [comment]
Yes, I intentionally only did Rel-16 (without OpenAPI) for now, to see where we end up in the discussions...

**Comment by gheikkila** [comment]
We could consider to also explicitly state in 11.4.2 that `apiRoot` shall be substituted by (one of) the `serverAddresses`. Currently this is sort of just understood between the lines...

**Comment by rjb1000** [comment]
Contributions to 3GPP **SA4#122** addressing this issue:
* TS 26.512:
  * Rel-16: **CR0031** in [S4-230041](https://www.3gpp.org/ftp/TSG_SA/WG4_CODEC/TSGS4_122_Athens/Docs/S4-230041.zip).
  * Rel-17: **CR0032** in [S4-230042](https://www.3gpp.org/ftp/TSG_SA/WG4_CODEC/TSGS4_122_Athens/Docs/S4-230042.zip).

**Comment by rjb1000** [comment]
Contributions revised and approved at 3GPP **SA4#122** Closing Plenary:
* TS 26.512:
  * Rel-16: **CR0031r1** in [S4-230266](https://www.3gpp.org/ftp/TSG_SA/WG4_CODEC/TSGS4_122_Athens/Docs/S4-230266.zip).
  * Rel-17: **CR0032r1** in [S4-230267](https://www.3gpp.org/ftp/TSG_SA/WG4_CODEC/TSGS4_122_Athens/Docs/S4-230267.zip).

**Comment by rjb1000** [comment]
CR pack [SP-250253](https://portal.3gpp.org/ngppapp/CreateTdoc.aspx?mode=view&contributionUid=SP-230253) approved at SA#99 Plenary meeting in Rotterdam.
